tests/regressions/pattern_match_heap_array.c3

```diff
@@ -9,4 +9,6 @@ function main : () -> int = {
 /// @COMPILE; EXPECT ERR
 /// File '*.c3', line 5, in 'function main: () -> int'
 /// Error: overload resolution for function call 'puts(u8[[]&[]])' failed
-/// No overloads available
+/// Available overloads:
+/// - function puts: (Span<u8>) -> int
+/// - function puts: (CString) -> int
```

Comment:
Wait, this looks like a pretty big change.  Are we now offering to initialize a struct implicitly, without wrapping the argument in `{ }`?

Reply:
No, we still require manual initialization. These are just the new (non-templated) overloads specified in `io.c3`.

But also this is probably a bug tbh, we should also list the generic functions in the list of available overloads.

Reply:
Oh right, duh. But you're right, we should be listing the generic function too.